ansible/roles/atmo-user-boot-scripts/tasks/main.yml

```diff
@@ -1,38 +1,71 @@
+- fail: msg="Variable DEPLOY_SCRIPTS is not set."
+  when: DEPLOY_SCRIPTS is not defined
+
+- fail: msg="Variable ASYNC_SCRIPTS is not set."
+  when: ASYNC_SCRIPTS is not defined
+
 - name: bootscript_1 - make directories for post scripts and logs
   file: path={{ item }} state=directory
-  with_items: '{{ POST_MAKE_DIR }}'
-
-- name: bootscript_2 - touch log files
-  file: path={{ item }} state=touch
-  with_items: '{{ POST_SCRIPT_LOGS }}'
-  # The following two lines work around https://github.com/ansible/ansible-modules-core/issues/170
-  register: touch_log
-  changed_when: touch_log.diff.before.state != "file"
-
-- name: bootscript_3 - run user's scripts in script directory
-  shell: >
-    for SCRIPT in {{ POST_SCRIPT_DIR_RUN }};
-    do
-        if [ -f $SCRIPT -a -x $SCRIPT ]
-        then
-            ATMO_USER={{ ATMOUSERNAME }} $SCRIPT 2>> {{ POST_SCRIPT_LOG_STDERR }} >> {{ POST_SCRIPT_LOG_STDOUT }}
-        fi
-    done
+  with_items: '{{ POST_MAKE_DIRS }}'
 
-- name: bootscript_4.1 - copy scripts to location
+- name: bootscript_2 - copy scripts to script directory
   copy:
     content: "{{ item.text }}"
-    dest: "{{POST_SCRIPT_DIR}}/{{item.name}}"
+    dest: "{{INSTANCE_SCRIPT_DIR}}/{{item.name}}"
     mode: "u+rwx"
-  with_items: '{{ SCRIPTS }}'
+  with_items: '{{ DEPLOY_SCRIPTS }}'
 
-- name: bootscript_4.2 - run scripts passed in to the playbook
-  shell: >
-    ATMO_USER={{ ATMOUSERNAME }} "{{POST_SCRIPT_DIR}}/{{ item.name }}" 2>> {{ POST_SCRIPT_LOG_STDERR }} >> {{ POST_SCRIPT_LOG_STDOUT }}
-  with_items: '{{ SCRIPTS }}'
+- name: bootscript_3 - Run deploy boot scripts synchronously, register results and return if failure occurs.
+  command: "{{INSTANCE_SCRIPT_DIR}}/{{item.name}}"
+  register: "deploy_script_result"
+  with_items: '{{ DEPLOY_SCRIPTS }}'
 
-- name: bootscript_4.3 - remove scripts
+- name: bootscript_4 - Remove instance scripts to avoid being picked up on next run
   file:
-    path: "{{POST_SCRIPT_DIR}}/{{ item.name }}"
+    path: "{{INSTANCE_SCRIPT_DIR}}/{{ item.name }}"
     state: absent
-  with_items: '{{ SCRIPTS }}'
+  with_items: '{{ DEPLOY_SCRIPTS }}'
+
+
+- name: bootscript_5 - copy scripts to script directory
+  copy:
+    content: "{{ item.text }}"
+    dest: "{{INSTANCE_SCRIPT_DIR}}/{{item.name}}"
+    mode: "u+rwx"
+  with_items: '{{ ASYNC_SCRIPTS }}'
+
+#Future TODO:
+# - If possible, refactor these components into ansible:
+#   - get name of  each file in script directory
+#   - if file is executable, generate 'dynamic log files' based on 'script|basename' and 'date()'
+#   - if file is executable, run ` set -m; nohup-call-to-file..` and point to dynamic log files from above.
```

Comment:
Drawing attention to this TODO for anyone aspiring to make such a change. Otherwise, the code and can be used as-is.